plugins/jobs/src/js/pages/JobDetailPageContainer.js

```diff
@@ -0,0 +1,110 @@
+import React from "react";
+import { componentFromStream, graphqlObservable } from "data-service";
+import { Observable, BehaviorSubject } from "rxjs";
+import gql from "graphql-tag";
+import { default as schema } from "#PLUGINS/jobs/src/js/data/JobModel";
+import Loader from "#SRC/js/components/Loader";
+import Page from "#SRC/js/components/Page";
+import RequestErrorMsg from "#SRC/js/components/RequestErrorMsg";
+import JobDetailPage from "./JobDetailPage";
+
+export const DIALOGS = {
+  EDIT: "edit",
+  DESTROY: "destroy"
+};
+
+const LoadingScreen = function() {
+  return (
+    <Page>
+      <Page.Header breadcrumbs={[]} />
+      <Loader />
+    </Page>
+  );
+};
+
+const ErrorScreen = function() {
+  return (
+    <Page>
+      <Page.Header breadcrumbs={[]} />
+      <RequestErrorMsg />
+    </Page>
+  );
+};
+
+const disabledDialog$ = new BehaviorSubject(null);
+const disableDialog = () => {
+  disabledDialog$.next(DIALOGS.DESTROY);
+};
+
+const jobActionDialog$ = new BehaviorSubject(null);
+
+const handleDestroyButtonClick = () => {
+  jobActionDialog$.next(DIALOGS.DESTROY);
+};
+
+const handleEditButtonClick = () => {
+  jobActionDialog$.next(DIALOGS.EDIT);
+};
+
+const closeDialog = () => {
+  jobActionDialog$.next(null);
+  disabledDialog$.next(null);
+};
+
+const getGraphQL = id =>
+  graphqlObservable(
+    gql`
+      query {
+        job(id: $id) {
+          id
+          name
+          path
+          command
+          schedules
+          cpus
+          description
+          mem
+          docker
+          disk
+          labels
+          jobRuns
+          json
+        }
+      }
+    `,
+    schema,
+    { id }
+  );
+
+export default componentFromStream(props$ => {
+  const id$ = props$.map(props => props.params.id).distinctUntilChanged();
+  const job$ = id$
+    .switchMap(id => getGraphQL(id))
+    .map(({ data: { job } }) => job);
+
+  return Observable.combineLatest([
+    job$,
+    props$,
+    jobActionDialog$,
+    disabledDialog$
+  ])
+    .map(([job, props, jobActionDialog, disabledDialog]) => (
+      <JobDetailPage
+        {...props}
+        job={job}
```

Comment:
this is the new job, correct? I cant find a place where this is converted into old struct, but UI code is working on the struct… can you clarify please? 😕

Reply:
Yes it is, the place where it's converted is the modal

Reply:
looks like you addressed the legacy parts 👍